sympy/polys/rings.py

```diff
@@ -2532,6 +2567,104 @@ def compose(f, x, a=None):
 
         return poly
 
+    def primitive_wrt(self, x):
+        """
+        Returns the content and primitive part of a polynomial with respect to a
+        specified variable.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x, y = ring("x, y", ZZ)
+        >>> p = 6*x**2*y - 9*x**3*y**2 + 3*x*y
+        >>> p.primitive_wrt(x)
+        (-9*x**3*y**2 + 6*x**2*y + 3*x*y, 1)
+
+        >>> p.primitive() # Difference between primitive and primitive_wrt results
+        (3, -3*x**3*y**2 + 2*x**2*y + x*y)
+        """
+        p = self
+        coeffs = p.coeff_split({x})
+        cont = gcd_extract(coeffs)
+        prim = p.exquo(cont)
+        return cont, prim
+
+    def _coeff_split_syms(self, syms):
+        """
+        Reorganize a polynomial over multiple variables into a polynomial over a
+        subset of those variables with coefficients being polynomials over the
+        remaining variables.
+        """
+        p1 = self
+        p2 = defaultdict(dict)
+        r = range(len(p1.ring.gens))
+
+        # Iterate through the terms and coefficients of the input polynomial
+        for m1, c1 in p1.items():
+            sym_indices = set(compress(r, m1))
+            m21 = [0] * len(r)
+            m22 = [0] * len(r)
+
+            # Separate variables into symbol and non-symbol categories
+            for i in sym_indices & syms:
+                m21[i] = m1[i]
+            for i in sym_indices - syms:
+                m22[i] = m1[i]
+
+            key1 = tuple(m21)
+            key2 = tuple(m22)
+
+            if key1 not in p2:
+                p2[key1] = {}
+
+            # Store the coefficient in the organized structure
+            p2[key1][key2] = p2.get(key1, {}).get(key2, 0) + c1
+
+        return p2
+
+    def coeff_split(self, syms):
+        """
+        Get the coefficients of a polynomial with respect to the specified
+        ``syms``.
+
+        For example, given a polynomial in ``p`` in ``K[x,y,z,t]``, ``p.
+        coeff_split({y, t})`` converts ``p`` to an element of ``K[x, z][y, t]``
+        and returns the coefficients as elements of ``K[x, z]``
+
+        Parameters
+        ==========
+
+        syms : set
+                A set of indices representing symbol variables.
+
+        Returns
+        =======
+
+        list
+            A list of polynomials resulting from splitting the input
+            polynomial.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x, y, z = ring("x, y, z", ZZ)
+
+        >>> f = 2*x**4 + 3*y**4 + 10*z**2 + 10*x*z**2
+        >>> syms = {2}
+        >>> f.coeff_split(syms)
+        [2*x**4 + 3*y**4, 10*x + 10]
```

Comment:
This would be easier to understand if the generator could be passed in rather than an integer.